### PR TITLE
Add Debug derives where needed and derive serde traits on Config

### PR DESCRIPTION
**Pull request by bschwind**
@skywhale please review

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -263,7 +275,7 @@ impl From<Config> for ffi::Config {
 }
 
 /// Statistics about the processor state.
-#[derive(Clone, Debug)]
+#[derive(Debug, Clone)]
```
Comment (skywhale) [inline src/config.rs:278]: maybe less confusing to derive serde for all the public structs?

File: src/config.rs [hunk 2]
```diff
@@ -147,7 +155,8 @@ impl From<NoiseSuppressionLevel> for ffi::NoiseSuppression_SuppressionLevel {
 }
 
 /// Noise suppression configuration.
-#[derive(Clone)]
+#[derive(Debug, Copy, Clone, PartialEq)]
```
Comment (skywhale) [inline src/config.rs:158]: Remove `Copy`?

**Comment by bschwind** [comment]
> I think you can get rid of the regexp code in -sys/build.rs, too.

Yeah I wasn't sure about that, in the case where someone might want to use the `sys` crate themselves but still have serde. I kinda doubt that would be common, so I'm okay with removing it if you are.

File: src/config.rs (hunk 2 above)
Comment (bschwind) [reply src/config.rs:158]: Good catch, I was just starting to place it wherever 😄

**Comment by bschwind** [comment]
@skywhale please take another look

**Pull request merged**